src/ucp/wireup/address.c

```diff
@@ -596,8 +597,10 @@ static ucs_status_t ucp_address_do_pack(ucp_worker_h worker, ucp_ep_h ep,
                                       context->tl_rscs[rsc_index].tl_name_csum);
 
             /* Transport information */
-            attr_len = ucp_address_pack_iface_attr(worker, ptr, rsc_index, iface_attr,
-                                                   worker->atomic_tls & UCS_BIT(rsc_index));
+            enable_amo = !(flags & UCP_ADDRESS_PACK_FLAG_DISABLE_HW_AMO) &&
```

Comment:
this is wrong: we can allow HW AMO for transport which have CPU atomics like shared memory, or devices with PCI atomics.
Instead, need to pass atomic_tls bitmap to address pack funciton

Reply:
Should we use the `atomic_tls` instead of `worker->atomic_tls` or do `atomic_tls & worker->atomic_tls` to calculate `enable_amo`?
Seems like we can calculate it without extra parameter just checking if any from `worker->atomic_tls` has `UCT_IFACE_FLAG_ATOMIC_CPU` capability and compare result to iface capabilities from `tl_bitmap` taking into account the fact that `worker->atomic_tls` is built only on 1 type of ifaces with cpu- or device-capable atomics, i.e. `UCT_IFACE_FLAG_ATOMIC_CPU` and `UCT_IFACE_FLAG_ATOMIC_DEVICE` can't be present on the same iface, right?

Reply:
Use it instead. Maybe we can add pack flag like ENABLE_DEVICE_ATOMICS. By default it's on. In sockaddr it will be off.

Reply:
Ok, let's fix it separately after https://github.com/openucx/ucx/pull/4277 is merged to avoid conflicts.

Reply:
fixed by https://github.com/openucx/ucx/pull/4304